src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -58,82 +66,128 @@ def _optional_identify_tool(
+                key_event.key() in (Qt.Key_Delete, Qt.Key_Backspace)
+                and point_count == 1
+            ):
+                # self.undo() resists to undo the first point so force back to initial
+                # reshape mode
+                self.stopCapturing()
+                self._change_to_reshape_mode_for_geom(
+                    self._common_segment_to_reshape, self._active_layer
+                )
 
     def canvasReleaseEvent(self, mouse_event: QgsMapMouseEvent) -> None:  # noqa: N802
-        location = self.toMapCoordinates(mouse_event.pos())
-        self._handle_mouse_click_event(location, mouse_event.button())
+        if self._tool_mode == ToolMode.PICK_SEGMENT:
+            if mouse_event.button() == Qt.LeftButton:
+                location = self.toMapCoordinates(mouse_event.pos())
+                self._handle_pick_segment_left_click(location)
+            return
+        super().canvasReleaseEvent(mouse_event)
 
-    @log_if_fails
-    def _handle_mouse_click_event(
-        self, location: QgsPointXY, mouse_button: Qt.MouseButton
-    ) -> None:
-        if self.tool_mode == ToolMode.PICK_SEGMENT and mouse_button == Qt.LeftButton:
-            self._handle_pick_segment_left_click(location)
-        elif self.tool_mode == ToolMode.RESHAPE:
-            if mouse_button == Qt.LeftButton:
-                self._handle_reshape_left_click(location)
-            elif mouse_button == Qt.RightButton:
+    def cadCanvasReleaseEvent(self, mouse_event: QgsMapMouseEvent) -> None:  # noqa N802
+        """Override of the QgsMapToolAdvancedDigitizing.cadCanvasReleaseEvent
+
+        This will receive adapted events from the cad system whenever a
+        canvasReleaseEvent is triggered and it's not hidden by the cad's
+        construction mode.
+
+        Args:
+            mouse_event (QgsMapMouseEvent): Mouse events prepared by the cad system
+        """
+
+        if self._tool_mode == ToolMode.RESHAPE:
+            if mouse_event.button() == Qt.LeftButton:
+                result = self.addVertex(
+                    mouse_event.mapPoint(), mouse_event.mapPointMatch()
+                )
+                if result == AddVertexReturn.TransformationError:
+                    MsgBar.warning(
+                        tr("Cannot transform the point to the layers coordinate system")
+                    )
+                    return
+                self.startCapturing()
+                self.start_point_indicator_rubber_band.hide()
+            elif mouse_event.button() == Qt.RightButton:
                 self._handle_reshape_right_click()
 
+    def cadCanvasMoveEvent(self, mouse_event: QgsMapMouseEvent) -> None:  # noqa N802
+        if self._tool_mode == ToolMode.RESHAPE and self.size() == 0:
+            self.start_point_indicator_rubber_band.movePoint(mouse_event.mapPoint())
+        return super().cadCanvasMoveEvent(mouse_event)
+
     def _handle_pick_segment_left_click(self, location: QgsPointXY) -> None:
```

Comment:
I added @log_if_fails now to `canvasReleseEvent` so it should catch all the errors. I still want the message triggered on https://github.com/nlsfi/segment-reshape-qgis-plugin/blob/use-digitizing-map-tool/src/segment_reshape/map_tool/segment_reshape_tool.py#L233 to be handled manually since log_if_fails shows the error with exception severity.